### PR TITLE
[Chore] clean up typos

### LINKED ISSUE #2160: feat: enable `codespell` in the lint workflow
### Motivation
The current workflow doesn't check the code spelling, which will allow some naive spelling errors in the commits. To reduce this situation, we would like to have a workflow to ensure the files in a PR are verified by `codespell.`

Please notice that some keywords will trigger false positives, so we have to maintain an ignore list.

---

I would suggest using codespell. Although it has some false positives, it can be suppressed via the `--ignore-words` file. The ability to misspell is limited.

Anyway, both tools have the option to fix typos automatically.

_Originally posted by @RRobot-lm in https://github.com/WasmEdge/WasmEdge/issues/2157#issuecomment-1346555798_

### PR DESCRIPTION
**Pull request by RRobot-lm**
Close #2160

### CODE REVIEW

**Comment by RRobot-lm** [comment]
Look like the CI failure is not caused by this change:
https://github.com/WasmEdge/WasmEdge/actions/runs/3718278759/jobs/6306367881
```
Package lib32gcc1 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  lib32gcc-s1

E: Package 'lib32gcc1' has no installation candidate
```

File: include/ast/section.h [hunk 1]
```diff
@@ -273,7 +273,7 @@ class AOTSection {
   /// @{
   uint32_t Version;
   uint8_t OSType;
-  uint8_t ArchType;
+  uint8_t archetype;
```
Comment (dannypsnl) [inline include/ast/section.h:276]: Is this expected?

File: include/common/errinfo.h [hunk 2]
```diff
@@ -172,8 +172,8 @@ struct InfoMismatch {
   }
 
   /// Case 5: unexpected external types
-  InfoMismatch(const ExternalType ExpExt, const ExternalType GotExt) noexcept
-      : Category(MismatchCategory::ExternalType), ExpExtType(ExpExt),
+  InfoMismatch(const ExternalType expect, const ExternalType GotExt) noexcept
```
Comment (dannypsnl) [inline include/common/errinfo.h:175]: I think this renaming didn’t make sense

File: include/common/filesystem.h [hunk 3]
```diff
@@ -1,7 +1,8 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: 2019-2022 Second State INC
 
-//===-- wasmedge/common/filesystem.h - std::filesystem seletion -----------===//
+//===-- wasmedge/common/filesystem.h - std::filesystem selection
+//-----------===//
```
Comment (dannypsnl) [inline include/common/filesystem.h:5]: Not expected formatting

File: lib/validator/formchecker.cpp [hunk 4]
```diff
@@ -38,7 +38,7 @@ void FormChecker::reset(bool CleanGlobal) {
     Tables.clear();
     Mems = 0;
     Globals.clear();
-    Datas.clear();
+    Data.clear();
```
Comment (dannypsnl) [inline lib/validator/formchecker.cpp:41]: I would say this breaks the name convention here, maybe need more consideration

File: plugins/wasi_crypto/asymmetric_common/func.cpp [hunk 5]
```diff
@@ -199,9 +199,9 @@ Expect<uint32_t> KeypairId::body(const Runtime::CallingFrame &Frame,
       unlikely(!Res)) {
     return Res.error();
   } else {
-    auto [ResSize, ResVersion] = *Res;
+    auto [resize, ResVersion] = *Res;
```
Comment (dannypsnl) [inline plugins/wasi_crypto/asymmetric_common/func.cpp:202]: Incorrect

File: lib/validator/formchecker.cpp (hunk 4 above)
Comment (RRobot-lm) [reply lib/validator/formchecker.cpp:41]: `data` is the plural form of `datum`. So we don't need to use `datas` to represent multiple data.

File: include/common/variant.h [hunk 6]
```diff
@@ -142,7 +142,7 @@ template <typename... Types> class Variant {
   static_assert(!(std::is_void_v<Types> || ...),
                 "variant must have no void alternative");
   static_assert((std::is_trivially_copyable_v<Types> && ...),
-                "variant must be trivially copyable");
+                "variant must be trivially copiable");
```
Comment (hydai) [inline include/common/variant.h:145]: It seems like this doesn't need to change: https://en.wiktionary.org/wiki/copyable#English

File: include/ast/section.h [hunk 7]
```diff
@@ -239,8 +239,8 @@ class AOTSection {
   void setOSType(uint8_t Type) noexcept { OSType = Type; }
 
   /// Getter and setter of arch type.
-  uint8_t getArchType() const noexcept { return ArchType; }
-  void setArchType(uint8_t Type) noexcept { ArchType = Type; }
+  uint8_t getArchType() const noexcept { return archetype; }
```
Comment (hydai) [inline include/ast/section.h]: This should not be changed.

File: docs/witx/typenames.witx [hunk 8]
```diff
@@ -44,7 +44,7 @@
     ;;; Argument list too long.
     $2big
     ;;; Permission denied.
-    $acces
+    $access
```
Comment (hydai) [inline docs/witx/typenames.witx:47]: This should be ignored: https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/witx/typenames.witx#L443-L455

File: docs/witx/typenames.witx [hunk 9]
```diff
@@ -461,7 +461,7 @@
 (typename $oflags
   (flags (@witx repr u16)
     ;;; Create file if it does not exist.
-    $creat
```
Comment (hydai) [inline docs/witx/typenames.witx:464]: This should be ignored:
https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/witx/typenames.witx#L443-L455

File: docs/book/zh/src/frameworks/mesh/dapr.md [hunk 10]
```diff
@@ -245,12 +245,12 @@ func imageHandlerWASI(_ context.Context, in *common.InvocationEvent) (out *commo
   vm.Validate()
   vm.Instantiate()
   res, err := vm.ExecuteBindgen("infer", wasmedge.Bindgen_return_array, image)
-  ans := string(res.([]byte))
+  and := string(res.([]byte))
```
Comment (hydai) [inline docs/book/zh/src/frameworks/mesh/dapr.md:248]: This should not be changed.

**Review by hydai — CHANGES_REQUESTED** [review]

File: lib/validator/formchecker.cpp (hunk 4 above)
Comment (hydai) [reply lib/validator/formchecker.cpp:41]: `Data` here means `DataSegment.` So it's fine to use `Datas`.
We remove all of the `Segment` to simplify the naming.

File: docs/book/zh/src/frameworks/mesh/dapr.md [hunk 11]
```diff
@@ -250,7 +250,7 @@ func imageHandlerWASI(_ context.Context, in *common.InvocationEvent) (out *commo
   vm.Delete()
   conf.Delete()
   out = &common.Content{
-    Data:    []byte(ans),
+    Data:    []byte(and),
```
Comment (hydai) [inline docs/book/zh/src/frameworks/mesh/dapr.md:253]: This should not be changed.

**Review by hydai — CHANGES_REQUESTED** [review]

File: lib/host/wasi/linux.h [hunk 12]
```diff
@@ -513,8 +513,8 @@ inline constexpr __wasi_protocol_t fromProtocal(int Protocal) noexcept {
   }
 }
 
-inline constexpr int toProtocal(__wasi_protocol_t Protocal) noexcept {
-  switch (Protocal) {
+inline constexpr int toProtocal(__wasi_protocol_t Protocol) noexcept {
+  switch (Protocol) {
```
Comment (dannypsnl) [inline lib/host/wasi/linux.h:517]: Functions' name didn't get fixed.

File: lib/host/wasi/macos.h [hunk 13]
```diff
@@ -453,8 +453,8 @@ inline constexpr int toSockType(__wasi_sock_type_t SockType) noexcept {
   }
 }
 
-inline constexpr __wasi_protocol_t fromProtocal(int Protocal) noexcept {
-  switch (Protocal) {
+inline constexpr __wasi_protocol_t fromProtocal(int Protocol) noexcept {
```
Comment (hydai) [inline lib/host/wasi/macos.h:456]: This should be `fromProtocol`.

File: lib/host/wasi/win.h [hunk 14]
```diff
@@ -669,8 +669,8 @@ inline constexpr int toSockType(__wasi_sock_type_t SockType) noexcept {
   }
 }
 
-inline constexpr __wasi_protocol_t fromProtocal(int Protocal) noexcept {
-  switch (Protocal) {
+inline constexpr __wasi_protocol_t fromProtocal(int Protocol) noexcept {
```
Comment (hydai) [inline lib/host/wasi/win.h:672]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/ignore_words [hunk 15]
```diff
@@ -0,0 +1,16 @@
+ans
+archtype
+copyable
+crate
+datas
+expext
```
Comment (dannypsnl) [inline .github/workflows/ignore_words:6]: Why this is an exception? Do you know where use this word?

Comment (RRobot-lm) [reply .github/workflows/ignore_words:6]: It is used to ignore https://github.com/WasmEdge/WasmEdge/pull/2165#discussion_r1051371780
codespell's ignore words only support all lowercase words so `expext` actually means `ExpExt`

Comment (dannypsnl) [reply .github/workflows/ignore_words:6]: Get it, can’t we write “ExpExt” here?

**Review by hydai — DISMISSED** [review]
Sounds like everything is good now. Should we get this merged? cc @dannypsnl @q82419

**Comment by hydai** [comment]
Oh, I found that there are some conflicting files. @RRobot-lm, Could you please fix them before we merge this?

**Review by dannypsnl — DISMISSED** [review]

**Comment by RRobot-lm** [comment]
> Oh, I found that there are some conflicting files. @RRobot-lm, Could you please fix them before we merge this?

Done

**Review by hydai — APPROVED** [review]

**Pull request merged**